opencsp/app/sofast/lib/SystemSofastFringe.py

```diff
@@ -246,54 +252,65 @@ def _measure_sequence_capture(
             # Run next operation if finished
             run_next()
 
-    def set_fringes(self, fringes: Fringes, min_display_value: int = None) -> None:
+    def set_calibration(self, calibration: ImageCalibrationAbstract) -> None:
         """
-        Loads fringe object and creates RGB fringe images to display.
-
-        Uses the global ImageProjection projector instance as the reference for scale.
+        Loads calibration object and creates RGB fringe images to display.
 
         Parameters
         ----------
-        fringes : Fringes
-            Fringe object to display.
-        min_display_value : int, None
-            Minimum display value to project. Should be based on the sensitivity of the camera in the optical system. In
-            a perfect system where the camera can tell the difference between the displayed values '0' and '1', this
-            value will be '0'. If None, then the value is retrieved from
-            self.calibration.calculate_min_display_camera_values().
+        calibration : ImageCalibrationAbstract
+            The image calibration object to apply to fringe image generation
+        """
+        self._calibration = calibration
 
+        self.create_fringe_images_from_image_calibration()
+
+    @property
+    def calibration(self):
+        """Returns ImageCalibration object. Note: this is not a setter, use set_calibration() instead"""
+        return self._calibration
+
+    def create_fringe_images_from_image_calibration(self):
+        """Once the system loads a new calibration object, a new min_display_value and
+        RGB fringe images must be made
         """
-        image_projection = ImageProjection.instance()
+        # Check fringes are loaded
+        if self.fringes is None:

```

Comment:
Does this require the fringes to have already been set before setting the calibration? If it's ok with you, I think it would be nice to be able to set the fringes and calibration in either order. So, either this:

```python
sys.calibration = calibration
sys.fringes = fringes
```

Or this:

```python
sys.fringes = fringes
sys.calibration = calibration
```

should both work fine.

Reply:
Good point! I modified `set_calibration()` and `set_fringes()` as follows. Let me know what you think.
```Python
def set_fringes(self, fringes: Fringes) -> None:
    """Sets the fringes object in the class
    
    Parameters
    ----------
    fringes : Fringes
        Fringe object to set to System
    """
    self.fringes = fringes
    
    if self._calibration is not None:
        self.create_fringe_images_from_image_calibration()

def set_calibration(self, calibration: ImageCalibrationAbstract) -> None:
    """
    Loads calibration object and creates RGB fringe images to display.

    Parameters
    ----------
    calibration : ImageCalibrationAbstract
        The image calibration object to use during fringe image generation
    """
    self._calibration = calibration

    if self.fringes is not None:
        self.create_fringe_images_from_image_calibration()
```